src/Host/Gimp/host_gimp.cpp

```diff
@@ -1187,9 +1195,24 @@ static GimpValueArray * gmic_qt_run(GimpProcedure * procedure, GimpRunMode run_m
     break;
   case GIMP_RUN_NONINTERACTIVE:
     gmic_qt_gimp_image_id = image;
+#if (GIMP_MAJOR_VERSION <= 2) && (GIMP_MINOR_VERSION <= 99) && (GIMP_MICRO_VERSION < 19)
     pluginParameters.command = g_value_get_string(gimp_value_array_index(args, 2));
     pluginParameters.inputMode = static_cast<GmicQt::InputMode>(g_value_get_int(gimp_value_array_index(args, 0)) + (int)GmicQt::InputMode::NoInput);
     pluginParameters.outputMode = static_cast<GmicQt::OutputMode>(g_value_get_int(gimp_value_array_index(args, 1)) + (int)GmicQt::OutputMode::InPlace);
+#else
+    char *command;
+    int inputMode;
+    int outputMode;
+    g_object_get(config,
+		 "command", &command,
+		 "input", &inputMode,
+		 "output", &outputMode,
+		 NULL);
+    pluginParameters.command = command;
+    pluginParameters.inputMode = (GmicQt::InputMode)inputMode;
+    pluginParameters.outputMode = (GmicQt::OutputMode)outputMode;
+    g_free(command);
```

Comment:
This looks wrong to me: by the time `pluginParameters.command` is used, the memory is already freed.

Reply:
The `pluginParameters` variable is an instance of `GmicQt::RunParameters` which is declared thus:

```
struct RunParameters {
  std::string command;
  std::string filterPath;
  InputMode inputMode = InputMode::Unspecified;
  OutputMode outputMode = OutputMode::Unspecified;
  std::string filterName() const;
};
```

My understanding is that assigning from `char *` to `std::string` will result in a deep copy, making it safe to release the original `char *` string.

Reply:
Ahh, didn’t catch that.